### PR TITLE
SAMA5D2 add new mcan ioctls

### PR DESCRIPTION
**Pull request by TimJTi**
## Summary
As per issue [here](https://github.com/apache/nuttx/issues/3927) and [this PR](https://github.com/apache/nuttx/pull/11061/commits) here are the changes I needed to make for the SAMA5D2 processor to have this all work as required.

## Impact
None to existing code.

## Testing
As per other PR, on custom board with SAMA5D2C-D1G and 2x CAN interfaces

### CODE REVIEW

**Review by acassis — APPROVED** [review]

**Comment by acassis** [comment]
@TimJTi nice work! Kudos!!!

**Review by xiaoxiang781216 — APPROVED** [review]

File: arch/arm/src/sama5/sam_mcan.c [hunk 1]
```diff
@@ -2838,6 +2842,70 @@ static int mcan_ioctl(struct can_dev_s *dev, int cmd, unsigned long arg)
...
+       *   Returned Value: Zero (OK) is returned on success.  Otherwise -1
+       *                   (ERROR) is returned with the errno variable set
+       *                   to indicate the nature of the error.
+       *   Dependencies:   None
+       */
+
+      case CANIOC_IFLUSH:
+        {
+          ret = mcan_cancel_rx_fifos(priv);
+          ret = mcan_cancel_tx_buffers(priv);
```
Comment (xiaoxiang781216) [inline arch/arm/src/sama5/sam_mcan.c:2873]: why flash tx buffer here?

Comment (TimJTi) [reply arch/arm/src/sama5/sam_mcan.c:2873]: Oops!

Comment (TimJTi) [reply arch/arm/src/sama5/sam_mcan.c]: fixed

**Pull request merged**